lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -46,7 +48,65 @@ mlir::tt::SystemDescAttr::getDefault(MLIRContext *context) {
           tt::ChipCoordAttr::get(context, 0, 0, 0, 0),
       },
       // Chip Channel Connections
-      {});
+      {
+          tt::ChipChannelAttr::get(context, 0, 0),
+      });
+}
+
+mlir::tt::SystemDescAttr
+mlir::tt::SystemDescAttr::getFromPath(MLIRContext *context, std::string &path) {
+  // Check if file exists
+  assert(!path.empty() && "cluster desc path must not be empty!");
+  std::ifstream fbb(path, std::ios::binary | std::ios::ate);
+  std::streampos size = fbb.tellg();
+  fbb.seekg(0, std::ios::beg);
+  auto buffer = std::shared_ptr<void>(std::malloc(size), std::free);
+  fbb.read(static_cast<char *>(buffer.get()), size);
+
+  // Read relevant information from binary
+  auto binary_system_desc =
+      ::tt::target::GetSizePrefixedSystemDescRoot(buffer.get())->system_desc();
+  auto const *binary_chip_desc = binary_system_desc->chip_descs();
+  auto const *binary_chip_desc_indices =
+      binary_system_desc->chip_desc_indices();
+  auto const *binary_chip_channels = binary_system_desc->chip_channels();
+  auto const *binary_chip_coords = binary_system_desc->chip_coords();
+
+  // Generate system desc attribute
+  auto system_desc_attr = tt::SystemDescAttr::get(
+      context,
+      {
+          tt::ChipDescAttr::get(
+              context, tt::ArchAttr::get(context, tt::Arch::WormholeB0),
+              {(*binary_chip_desc)[0]->grid_size()->y(),
+               (*binary_chip_desc)[0]->grid_size()->x()},
+              (*binary_chip_desc)[0]->l1_size(),
+              (*binary_chip_desc)[0]->num_dram_channels(),
+              (*binary_chip_desc)[0]->dram_channel_size(),
+              (*binary_chip_desc)[0]->noc_l1_address_align_bytes(),
+              (*binary_chip_desc)[0]->pcie_address_align_bytes(),
+              (*binary_chip_desc)[0]->noc_dram_address_align_bytes()),
+      },
```

Comment:
We should loop over `chip_descs` into a small vector of `tt::ChipDescAttr` and pass them in here.  Might as well support multi device here, I know other parts of the stack will fail, but let's code it for multi-device support from the start.